### ISSUE TITLE
[Bug] Flatpack does not prevent idle

### ISSUE THREAD

**Issue by BenMatteson** [Vencord/Vesktop#577]
### Operating System

Manjaro

### Linux Only ~ Desktop Environment

KDE X11

### What happens when the bug or crash occurs?

System idles while watching stream (screen dim/off, screensaver, etc.) when using flatpack version.

### What is the expected behaviour?

While watching a stream, system idle is prevented
Confirmed appimage works as expected.

### How do you recreate this bug or crash?

1. Ensure a system idle behavior is enabled (screen off, sleep, etc.)
2. Start Vesktop Flatpack image
3. Join a voice channel and watch another user's stream
4. wait for first idle behavior

### Debug Logs

```
Not sure how to start flatpack from terminal... I can look it up if needed, but Inhibit portal permission looks like what's needed (I just asked the internet, I don't know anything about creating flatpacks.)
```

### Request Agreement

- [X] I have searched the existing issues and found no similar issue
- [X] I am using the latest Vesktop and Vencord versions
- [X] This issue occurs on an official release (not just the AUR or Nix packages)

**Comment by powerofthe69** [comment]
I don't believe the Flatpak version of the app is officially supported by the Vesktop team. If the Appimage is working as your post suggested, it would do more good to open this ticket at the flatpak issue page: https://github.com/flathub/dev.vencord.Vesktop

**Comment by Vendicated** [comment]
> I don't believe the Flatpak version of the app is officially supported by the Vesktop team

it is!

**Comment by BenMatteson** [comment]
Vindicated by Vendicated!
Thanks for the command, I ran it and the output confirmed unrelated, but I'll drop it in anyway now that I have it.

**Comment by kostadinsh** [comment]
Can you check if enabling "Session Bus Access" and/or "System Bus Access" in KDE's flatpak permissions for Vesktop prevent it from idling?

**Comment by BenMatteson** [comment]
Neither worked, unless I need to restart between toggling them or something.

**Comment by BenMatteson** [comment]
Ok, I should probably have figured, but I reinstalled the flatpack due to some issue with updating vencord (I think that was on me) and it does work properly with both Session and System Bus access if I set them before launching.

**Comment by Vendicated** [comment]
for now you can use the CustomIdle plugin as a workaround

**Comment by Vendicated** [comment]
how Discord does it:
1. subscribe to ChannelRTCStore and SelectedChannelStore
2. whenever `SelectedChannelStore.getVoiceChannelId()` is not null and `ChannelRTCStore.getMode() === "video"`, it blocks display sleep: `DiscordNative.powerSaveBlocker.blockDisplaySleep()`

**Comment by jigyashu-bhatt** [comment]
will this work ?? 
const { ipcRenderer } = require('electron');

// When the stream starts
function onStreamStart() {
  ipcRenderer.invoke('stream-started').then((id) => {
    console.log('Stream started. Power saver blocker active.');
  });
}

// When the stream stops
function onStreamStop() {
  ipcRenderer.invoke('stream-stopped').then(() => {
    console.log('Stream stopped. Power saver blocker deactivated.');
  });
}

// Example usage with a video element
const videoElement = document.querySelector('video');

videoElement.addEventListener('play', onStreamStart);
videoElement.addEventListener('pause', onStreamStop);
videoElement.addEventListener('ended', onStreamStop);
